addons/website_event/models/event_event.py

```diff
@@ -499,6 +500,18 @@ def _get_event_resource_urls(self, slot=False):
             iCal_url += '?' + werkzeug.urls.url_encode({'slot_id': slot.id})
         return {'google_url': google_url, 'iCal_url': iCal_url}
 
+    def _get_event_open_slots(self):
```

Comment:
Would vote for a "_filter_open_slots" method on the event.slot model instead.

This logic is very specific to event slots which is a more niche configuration of events, doesn't seem super valuable to have on the event model itself.

It will behave the same inside the python / xml as you can do "event.event_slot_ids._filter_open_slots()".